source/text/Format.cpp

```diff
@@ -389,3 +417,103 @@ vector<string> Format::Split(const string &str, const string &separator)
 	}
 	return result;
 }
+
+
+
+string Format::ExpandConditions(const string &source, ConditionGetter getter)
+{
+	string result;
+	result.reserve(source.size());
+
+	size_t formatStart = string::npos;
+	size_t formatSize = string::npos;
+	size_t conditionStart = string::npos;
+	size_t conditionSize = string::npos;
+
+	// Hand-coded regular grammar parser for:
```

Comment:
we have non handcoded code here?
now I am interested

Reply:
I should use a regular expression library, or flex/bison, but such tools are not available to me.

Reply:
Actually, regular expressions aren't enough here, since there are nested parentheses, but another parser generator would do it just fine.

Reply:
Seriously, nobody should be *writing parsers*. There are many options for parser generators, and they're far less error-prone than a human.

Reply:
hmmm, ok, I did not know about these parsers, thank you for the info